lms/djangoapps/courseware/views/index.py

```diff
@@ -159,12 +159,15 @@ def _setup_masquerade_for_effective_user(self):
         be processed for the requested effective user.
         """
         self.real_user = self.request.user
-        self.masquerade, self.effective_user = setup_masquerade(
-            self.request,
-            self.course_key,
-            self.is_staff,
-            reset_masquerade_data=True
-        )
+        if self.view == PUBLIC_VIEW:
+            self.masquerade, self.effective_user = None, AnonymousUser()
+        else:
+            self.masquerade, self.effective_user = setup_masquerade(
+                self.request,
+                self.course_key,
+                self.is_staff,
+                reset_masquerade_data=True
+            )
```

Comment:
@ormsbee This PR is next in the series of work for access for unenrolled users. It adds the `public_view` for the video XBlock. Have a couple of questions here:

1. When a signed-in but unenrolled user visits a course open to public `self.request.user = self.effective_user` causes the template to show them as logged out. One option is that we undo this change and not explicitly set `self.effective_user = AnonymousUser()` for these users. Another is that we set `self.request.user = self.effective_user` only for `student_view` (i.e. in the else clause). What would the better option? Is there a third one?
2. I see you added the `read_only` parameter to `FieldDataCache.cache_for_descriptor_descendents()`. Should we set that for the public_view as well?

Thanks.

Reply:
When a signed-in-but-unenrolled user visits a video marked for public access, I think it's fine if they have their user state carry with them (i.e. not set them to anonymous).

The `read_only` flag is there to minimize lock contention. It was added because people would run crawlers with requests that would block each other while trying to update their sequence position in CSM. Since we might want to write CSM state for people who are logged in and unenrolled, I think it's fine if `public_view` does not use the `read_only` flag there.

Reply:
@ormsbee Thanks!

@pkulkark That means we can undo the changes in this file.